clients/vscode/src/code-action/QuickFix.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  CancellationToken,
+  CodeAction,
+  CodeActionContext,
+  CodeActionKind,
+  CodeActionProvider as CodeActionProviderInterface,
+  Range,
+  Selection,
+  TextDocument,
+} from "vscode";
+import { ContextVariables } from "../ContextVariables";
+import { getLogger } from "../logger";
+export class QuickFixCodeActionProvider implements CodeActionProviderInterface {
+  constructor(private readonly contextVariables: ContextVariables) {}
+
+  provideCodeActions(
+    _document: TextDocument,
+    _range: Range | Selection,
+    context: CodeActionContext,
+    token: CancellationToken,
+  ): CodeAction[] | undefined {
+    if (token.isCancellationRequested) {
+      return;
+    }
+
+    if (!this.contextVariables.chatEnabled) {
+      return;
+    }
+    if (context.diagnostics.length === 0) {
+      return [];
+    }
+
+    //TODO: Get all diagnostics from the context and substring with max command chars
+    const lspErrors = context.diagnostics[0];
+    getLogger("QuickFixCodeActionProvider").info("lspErrors", lspErrors);
+
+    const quickFixEditing = new CodeAction("Fixing with Tabby", CodeActionKind.QuickFix);
+    quickFixEditing.command = {
+      command: "tabby.chat.edit.start",
+      title: "Fix with Tabby",
+      arguments: [
+        {
+          errorContext: lspErrors,
+        },
```

Comment:
```suggestion
    userCommand
```

Construct userCommand here directly using lspErrors.

Reply:
Example prompt:

Here is some error information that occurred in the selection:

Error #1:  ...
Error #2: ...

Please correct the errors

```